src/stores/cookie.ts

```diff
@@ -68,33 +78,36 @@ export class CookieSessionStorage {
       value: await create(
         { alg: "HS512", typ: "JWT" },
         { ...session.data, _flash: session.flashedData },
-        this.#key,
+        this.#key
       ),
       path: "/",
+      ...this.#cookieOptions,
     });
 
     return response;
   }
 }
 
-export async function cookieSession(
-  req: Request,
-  ctx: MiddlewareHandlerContext<WithSession>,
-) {
-  const { sessionId } = getCookies(req.headers);
-  const cookieSessionStorage = await createCookieSessionStorage();
-
-  if (
-    sessionId && (await cookieSessionStorage.exists(sessionId))
+export function createCookieSession(cookieOptions?: CookieOptions) {
+  return async function cookieSession(
```

Comment:
```suggestion
export function cookieSession(cookieOptions?: CookieOptions) {
  return async function (
```

Using this will allow using `cookieSession` which makes it more concise

if we decide to go with it, it also needs to be reflected in the documentation.

Reply:
That is certainly true.
I'll reflect that.